aws/resource_aws_codecommit_approval_rule_association.go

```diff
@@ -0,0 +1,148 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/codecommit"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceAwsCodeCommitApprovalRuleAssociation() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsCodeCommitApprovalRuleAssociationCreate,
+		Update: resourceAwsCodeCommitApprovalRuleAssociationUpdate,
+		Read:   resourceAwsCodeCommitApprovalRuleAssociationRead,
+		Delete: resourceAwsCodeCommitApprovalRuleAssociationDelete,
+
+		Schema: map[string]*schema.Schema{
+			"template_name": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
```

Comment:
according to docs lets add length validation https://docs.aws.amazon.com/codecommit/latest/APIReference/API_AssociateApprovalRuleTemplateWithRepository.html